client/signup/controller.js

```diff
@@ -192,9 +192,13 @@ export default {
 		next();
 	},
 
-	saveInitialContext( context, next ) {
-		if ( ! initialContext ) {
-			initialContext = Object.assign( {}, context );
+	saveInitialContext( newContext, next ) {
+		if (
+			! initialContext ||
+			// Reset initialContext when site query param changes between old context and new.
+			( newContext?.query?.site && newContext.query.site !== initialContext?.query?.site )
```

Comment:
```suggestion
			( newContext?.query?.site &&  initialContext?.query?.site && newContext.query.site !== initialContext?.query?.site )
```

Thoughts?

Reply:
I think I may be partial to the first method.

Reply:
I worry this change might introduce unexpected behavior when other flows start using a `site` parameter in the future

Reply:
> my blind expectation is the context of the site should switch to the current or most recently seen parameter

Sure, but that's not what we're doing here. My worry is about introducing a special case for when a flow happens to use a parameter called `site`.

Reply:
Query params only show up in the initial request, after that they're stored in localstorage. I think its a bit of a trade off either way.